Realm/RLMObjectStore.mm

```diff
@@ -453,11 +454,12 @@ void RLMAddObjectToRealm(RLMObjectBase *object, RLMRealm *realm, RLMCreationOpti
 
 
 RLMObjectBase *RLMCreateObjectInRealmWithValue(RLMRealm *realm, NSString *className, id value, RLMCreationOptions options) {
-    if (RLMIsObjectSubclass([value class]) &&
-        [[[(RLMObjectBase *)value class] className] isEqualToString:className] &&
-        [(RLMObjectBase *)value realm] == realm) {
-        // This is a no-op if value is an RLMObject of the same type already backed by the target realm.
-        return value;
+    if (options & RLMCreationOptionsUpdateOrCreate && RLMIsObjectSubclass([value class])) {
```

Comment:
You should be able to make a copy of an object by passing it in as the value. The old code path simply returned the original object rather than copying when not in update mode.